### DISCUSSION TITLE
Suggestion for improving type inference

### DISCUSSION THREAD

**Discussion by Gruzchick** [TanStack/query#574] [General]
Hi! Before all, I am want to say a great job )!

I have a similar code:

`const fetchData = async (
  key: string,
  place: PlaceType,
) => {
  /* some code */
};

const place : PlaceType | null = useSelector(/* some code*/) // place in some cases may be "null" !

const info = useQuery({
      queryKey: place && ['currentWeather-${place.id}'],
      variables: [place],
      queryFn: fetchData
});`

And I get a type error because of the typescript infer that a null value can be passed to the function. But essentially, in the "queryKey" property, I made type guard. I'm not sure if this can be implemented, but it would be cool.

Although perhaps I did not understand how to use this right way )

**Comment by wolverineks** [comment]
`queryKey` cant be null or `false`
if you want to disable the query until you have `place`,
you can use something like:
```ts
const info = useQuery({
  queryKey: ['currentWeather', place?.id],
  queryFn: () => {
    // do something with place
  },
  config: { enabled: !!place }
});
```
